CefSharp.BrowserSubprocess.Core/CefAppUnmanagedWrapper.cpp

```diff
@@ -91,64 +91,105 @@ namespace CefSharp
     {
         auto handled = false;
         auto name = message->GetName();
-        if (name == kEvaluateJavascriptRequest)
+        auto argList = message->GetArgumentList();
+
+        auto browserId = argList->GetInt(0);
+        if (browser->GetIdentifier() != browserId)
         {
-            auto argList = message->GetArgumentList();
-            auto browserId = argList->GetInt(0);
-            auto frameId = GetInt64(argList, 1);
-            auto callbackId = GetInt64(argList, 2);
-            auto script = argList->GetString(3);
+            throw gcnew InvalidOperationException(String::Format("Request BrowserId : {0} does not match browser Id : {1}", browserId, browser->GetIdentifier()));
```

Comment:
We can do that, but what would that contain? I mean should we put the callbackid into it, so  we can fail the task on the other side? If so what happens for a message when there is no callback (like destroy callback or messages yet to add)? Should we have multiple different error messages? Or should there be a unique correlation id kind of thing in each and every message so we can report any kind of error? So my only concern is that it raises a set of problems in a context which shouldn't really happen. Anyway I'll try to cook up something tonight, let's see how that goes :)